lib/active_support/cache/database_store.rb

```diff
@@ -120,6 +131,25 @@ def delete_entry(key, _options = nil)
         @model.where(key: key).destroy_all
       end
 
+      def write_multi_entries(hash, **_options)
+        entries = hash.map {|key, entry| { key: key, created_at: Time.zone.now, **entry_attributes(entry) } }
+
+        # In rails 7, we can use update_only not to do anything. But for the sakes of compatibility, we don't use any additional parameters.
+        @model.upsert_all(entries)
+      end
+
+      def entry_attributes(entry)
+        expires_at = Time.zone.at(entry.expires_at) if entry.expires_at
+        attributes = case entry.value
+                     when Numeric # Integer and float entries do not warrant compression
+                       { compression: 'plain', value: compress(entry.value, 'plain') }
+                     else
+                       { compression: @compression, value: compress(entry.value, @compression) }
+                     end
+
+        attributes.merge(version: entry.version.presence, expires_at: expires_at)
```

Comment:
use `attributes.update`, we don't want to create a new object, update in place is fine